cpp/include/cudf/io/text/byte_range_info.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace cudf {
+namespace io {
+namespace text {
+
+/**
+ * @brief stores offset and size used to indicate a byte range
+ */
+struct byte_range_info {
+ public:
+  int64_t offset;
+  int64_t size;
+
+  byte_range_info() : offset(0), size(0) {}
+  byte_range_info(int64_t offset, int64_t size) : offset(offset), size(size) {}
```

Comment:
I guess my comment above would make a lot more sense if you also defined the struct like this:
```suggestion
  int64_t offset{0};
  int64_t size{0};
```
Then the constructors are implicitly handled too.